armoryengine/Transaction.py

```diff
@@ -430,6 +431,8 @@ def TxInExtractAddrStrIfAvail(txinObj):
       return hash160_to_addrStr( hash160(lastPush) )
    elif scrType == CPP_TXIN_SPENDP2SH:
       return binScript_to_p2shAddrStr(lastPush)
+   elif scrType == CPP_TXIN_P2WPKH_P2SH or scrType == CPP_TXIN_P2WSH_P2SH:
+      return binScript_to_p2shAddrStr(hash160(rawScript))
    else:
```

Comment:
It depends on what the txin is really. If it's a payment you performed, you would want to know it came from your wallets. In this case turning a wp2pkh scriptPubKey into the underlying b58 address has merits, i.e. check on code first, if it matches an address your wallets, display it at least he origin wallet id, otherwise leave it blank or just show the hash.